sys/console/full/src/console.c

```diff
@@ -116,7 +116,11 @@ console_read(char *str, int cnt, int *newline)
     }
     cmd = ev->ev_arg;
     len = strlen(cmd->line);
-    strncpy(str, cmd->line, len+1);
+    if (cnt < len) {
+        len = cnt;
```

Comment:
Maybe this would be better?
if (cnt -1 < len) {
   len = cnt - 1;
}

Thanks for noticing this, BTW.